src/main/java/sirius/biz/codelists/LookupTable.java

```diff
@@ -771,45 +784,90 @@ public Stream<LookupTableEntry> suggest(@Nullable String searchTerm) {
         }
-        return performSuggest(new Limit(0, MAX_SUGGESTIONS), searchTerm, language);
+        return performSuggest(new Limit(0, MAX_SUGGESTIONS), searchTerm, language, false);
+    }
+
+    /**
+     * Suggests several entries for the given search term using the given language.
+     *
+     * @param searchTerm               the term used to filter the suggestions
+     * @param language                 the language to translate the name and description to
+     * @param considerDeprecatedValues whether deprecated values should be suggested as well
+     * @return a stream of suggestions for the given term. Note that most probably {@link Stream#limit(long)} should
+     * be used on the result as this might yield quite a bunch of suggestions in order to optimize internal queries.
+     */
+    public Stream<LookupTableEntry> suggest(@Nullable String searchTerm,
+                                            String language,
+                                            boolean considerDeprecatedValues) {
+        if (Strings.isEmpty(searchTerm)) {
+            return scan(language, new Limit(0, MAX_SUGGESTIONS), considerDeprecatedValues);
+        }
+        return performSuggest(new Limit(0, MAX_SUGGESTIONS), searchTerm, language, considerDeprecatedValues);
     }
 
     /**
      * Computes the actual suggestions for the given query and limit.
      *
-     * @param limit      regulates how many suggestions are generated
-     * @param searchTerm the term to search by
-     * @param language   the language to translate names and descriptions in
+     * @param limit                    regulates how many suggestions are generated
+     * @param searchTerm               the term to search by
+     * @param language                 the language to translate names and descriptions in
+     * @param considerDeprecatedValues whether deprecated values should be suggested as well
      * @return a stream containing all suggestions
      */
-    protected abstract Stream<LookupTableEntry> performSuggest(Limit limit, String searchTerm, String language);
+    protected abstract Stream<LookupTableEntry> performSuggest(Limit limit,
+                                                               String searchTerm,
+                                                               String language,
+                                                               boolean considerDeprecatedValues);
 
     /**
-     * Enumerates all entries in the table using the current language.
+     * Enumerates all entries in the table aside from deprecated entries using the current language.
      *
      * @param limit the limit to apply to fetch a sane number of entries
      * @return a stream of all entries in this table, or an empty stream is scanning isn't supported
      */
     public Stream<LookupTableEntry> scan(Limit limit) {
-        return scan(NLS.getCurrentLanguage(), limit);
+        return scan(NLS.getCurrentLanguage(), limit, true);
```

Comment:
true? thought default was false...

Reply:
`true` is the default when it comes to all `scan()` methods as the implementations never filtered out deprecated values there.

Look at e.g. `sirius.biz.codelists.IDBLookupTable#scan(String, Limit)`. While it did mark deprecated entries as such, it never filtered them out. Same goes for the other implementations.